api/reviews_test.go

```diff
@@ -568,7 +587,260 @@ func TestAddReviewHandler(t *testing.T) {
 			api.addReviewHandler(response, request)
 			Convey("Then the HTTP response code is 400", func() {
 				So(response.Code, ShouldEqual, http.StatusBadRequest)
-				So(response.Body.String(), ShouldEqual, "invalid review. Missing required field\n")
+				So(response.Body.String(), ShouldEqual, "invalid review\n")
+			})
+		})
+	})
+}
+
+func TestUpdateReviewHandler(t *testing.T) {
+	Convey("Given an HTTP PUT request to the /books/{id}/reviews/{review_id} endpoint", t, func() {
+
+		Convey("When the book and review exist, and the review update is valid", func() {
+			// TODO - how can I get GetReviewFunc to return a different value depending on how many times it has been called?
```

Comment:
This can get a bit messy but you could keep track/maintain of an invocation count and implement something like the example below. Essentially you use the invocation count to determine which values to return.

https://play.golang.org/p/ckZkPJ5DeYS